### DISCUSSION TITLE
prisma.model.count() returns wrong number in combination with fullTextSearch

### DISCUSSION THREAD

**Discussion by nemanjam** [prisma/prisma#11575] [Q&A] [answered]
I use latest `3.9` client with Postgres and `previewFeatures = ["fullTextSearch"]` enabled, and `findMany()` works mostly fine and it does filtering, I get 6 items out of 24, but `count()` always returns total number of instances (24) like there isn't any filtering. And I pass exact same `where` filter to both `findMany()` and `count()`. I need correct number of filtered items for pagination to work correctly. This happens when I use `search` filter in `where`.

Here is the code:

```typescript
  const where = {
    where: {
      ...(searchTerm && {
        OR: [
          {
            title: {
              search: searchTerm,
            },
          },
          {
            content: {
              search: searchTerm,
            },
          },
          {
            author: {
              username: {
                search: searchTerm,
              },
            },
          },
          {
            author: {
              name: {
                search: searchTerm,
              },
            },
          },
        ],
      }),
    },
  };
```

Then I use exact same `where` object in both `findMany()` and `count()`:

```typescript
  const totalCount = await prisma.post.count({ ...where }); // returns 24 always, no filtering

  let posts = await prisma.post.findMany({ // returns 6 items
    ...where,
    include: {
      author: true,
    },
    orderBy: {
      updatedAt: sortDirection as SortDirectionType,
    },
  });
```

**Comment by nikolasburk** [comment, marked as answer]
Hey @nemanjam, thanks a lot for raising this! It seems like you are encountering the same bug that has been reported [here](https://github.com/prisma/prisma/issues/11556)? We already have a [fix](https://github.com/prisma/prisma/pull/11560) in the works that should be available soon! In the meantime, you should be able to downgrade to `v3.8.x` to work around the problem temporarily.

If you think you are encountering a different issue, let me know and we'll look into it more 🙂

**Reply by nemanjam** [reply]
I will use this until `count()` is not fixed.

```typescript
const totalCount = (await prisma.post.findMany({ ...where }))?.length || 0;`
```

**Reply by nemanjam** [reply]
I updated to `3.9.1` and it's mostly working now but I still get errors when call `count()` here and there.

```
my error PrismaClientUnknownRequestError: 
Invalid `prisma.post.count()` invocation:


  Error occurred during query execution:
ConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(Error { kind: Db, cause: Some(DbError { severity: "ERROR", parsed_severity: Some(Error), code: SqlState("42601"), message: "syntax error in tsquery: \"praesentium quia\"", detail: None, hint: None, position: None, where_: None, schema: None, table: None, column: None, datatype: None, constraint: None, file: Some("tsquery.c"), line: Some(689), routine: Some("makepol") }) }) })
    at cb (/home/username/Desktop/nextjs-prisma-boilerplate/node_modules/@prisma/client/runtime/index.js:38707:17)
    at async PrismaClient._request (/home/username/Desktop/nextjs-prisma-boilerplate/node_modules/@prisma/client/runtime/index.js:40853:18)
    at async getPostsWithAuthor (webpack-internal:///(api)/./pages/api/posts/index.ts:93:22)
    at async Array.eval (webpack-internal:///(api)/./pages/api/posts/index.ts:125:17) {
  clientVersion: '3.9.1'
}
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
